src/index-router/index-deposit-router.ts

```diff
@@ -0,0 +1,106 @@
+import { BigNumberish, BigNumber, ContractTransaction } from 'ethers'
+
+import { Account } from '../account'
+import { Contract } from '../contract'
+import {
+  IndexDepositRouter as IndexDepositRouterInterface,
+  IndexDepositRouter__factory,
+} from '../typechain'
+import { PromiseOrValue } from '../typechain/common'
+import { IReserveRouter } from '../typechain/IndexDepositRouter'
+import { Address, ChainId, ChainIds } from '../types'
+
+/** ### Default IndexDepositRouter address for network */
+export const defaultIndexDepositRouterAddress: Record<ChainId, Address> = {
+  /** ### Default IndexDepositRouter address on mainnet. */
+  [ChainIds.Mainnet]: '0x0',
+  /** ### Default IndexDepositRouter address on c-chain. */
+  [ChainIds.CChain]: '0xa04df6ec0138b9366c28d018d16acffd76531855', //INFO: CChain staging
+}
+
+/** ### IndexDepositRouter Contract */
+export class IndexDepositRouter extends Contract<IndexDepositRouterInterface> {
+  /**
+   * ### Creates a new IndexDepositRouter instance
+   *
+   * @param account Account to use for signing
+   * @param contract Contract instance or address of the IndexDepositRouter contract
+   *
+   * @returns New IndexDepositRouter token instance
+   */
+  constructor(
+    account: Account,
+    contract: IndexDepositRouterInterface | Address,
+  ) {
+    super(account, contract, IndexDepositRouter__factory)
+  }
+
+  /**
+   * ### Deposit
+   *
+   * @param index buy token address
+   * @param sellAmount sell token's  amount
+   * @param quotes (optional) quotes for deposits with USDC
+   *
+   * @returns deposit transaction
+   */
+  async deposit(
+    index: PromiseOrValue<string>,
+    sellAmount: BigNumberish,
+    params?: IReserveRouter.QuoteParamsStruct,
```

Comment:
maybe pass here a default param `gasPercentage = 5 or 95`?

Reply:
yeah, make sense

Reply:
@josipkoncurat Actually I've been thinking about that idea and it seems like we will not use it on front-end both as user input value or value set programmatically. Anyways the value shouldn't be greater than 5% at any conditions.

Reply:
Sometimes you will probably have to use it or having an easy way of passing it into the function just to test out something, so my suggestion is to have it as a parameter to a function with a default value of 5%.